From 21a8680b7bd436867ac689a0b4b1b18a3359e208 Mon Sep 17 00:00:00 2001
From: Geoff Parker <geoffrey.parker@arthrex.com>
Date: Fri, 9 Oct 2020 17:48:47 +0000
Subject: [PATCH] libdmmp/Makefile: replace perl with sed in install target to
 work with pseudo

The multipath-tools libdmmp/Makefile install target uses 'perl -i' instead
of 'sed -i' for string substitutions. The perl method creates a temporary file
and overwrites the original which changes the inodes and corrupts the pseudo
db. Changes to pseduo cause a build abort rather than allow possible bad
ownership or permissions settings on the files. 'sed -i' is compatible
with pseudo.

Signed-off-by: Geoff Parker <geoffrey.parker@arthrex.com>
[OP: Rebase to 0.9.3]
Signed-off-by: Ovidiu Panait <ovidiu.panait@windriver.com>
---
Upstream-Status: Pending

 libdmmp/Makefile | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/libdmmp/Makefile b/libdmmp/Makefile
index e4589250..34e06425 100644
--- a/libdmmp/Makefile
+++ b/libdmmp/Makefile
@@ -39,11 +39,11 @@ install:
 	$(LN) $(LIBS) $(DESTDIR)$(usrlibdir)/$(DEVLIB)
 	$(INSTALL_PROGRAM) -m 644 -D \
 		$(PKGFILE).in $(DESTDIR)$(pkgconfdir)/$(PKGFILE)
-	perl -i -pe 's|__VERSION__|$(LIBDMMP_VERSION)|g' \
+	sed -i 's|__VERSION__|$(LIBDMMP_VERSION)|g' \
 		$(DESTDIR)$(pkgconfdir)/$(PKGFILE)
-	perl -i -pe 's|__LIBDIR__|$(usrlibdir)|g' \
+	sed -i 's|__LIBDIR__|$(usrlibdir)|g' \
 		$(DESTDIR)$(pkgconfdir)/$(PKGFILE)
-	perl -i -pe 's|__INCLUDEDIR__|$(includedir)|g' \
+	sed -i 's|__INCLUDEDIR__|$(includedir)|g' \
 		$(DESTDIR)$(pkgconfdir)/$(PKGFILE)
 	$(INSTALL_PROGRAM) -d 755 $(DESTDIR)$(man3dir)
 	$(INSTALL_PROGRAM) -m 644 -t $(DESTDIR)$(man3dir) docs/man/*.3
-- 
2.38.1

